include/libtorrent/bencode.hpp

```diff
@@ -90,7 +90,7 @@ namespace libtorrent
 			= typename std::enable_if<std::is_integral<In>::value>::type>
 		int write_integer(OutIt& out, In data)
 		{
-			entry::integer_type const val(data);
+			entry::integer_type const val = entry::integer_type(data);
```

Comment:
(I did not) I needed to restore this because when I compiled in my warning free branch, this direct construction didn't count like an explicit cast, only doing this the warning goes away.